### ISSUE TITLE
godaddy API Change?

### ISSUE THREAD

**Issue by jakeapp** [eXamadeus/godaddypy#3]
Response Data: {
    u'fields':  [  { 
        u'path': u'records', 
        u'message': u'is not a array', 
        u'code': u'UNEXPECTED_TYPE'
     }  ], 
     u'message': u"Request body doesn't fulfill schema, see details in `fields`", 
     u'code': u'INVALID_BODY'
}

**Comment by qigtang** [comment]
seems changed, I need to manually change code to make it working again.

--- /usr/local/lib/python2.7/dist-packages/godaddypy/client.py  2018-04-20 22:03:13.466927835 -0700
+++ /usr/local/lib/python2.7/dist-packages/godaddypy/client.py.bak      2018-04-20 21:58:39.783297049 -0700
@@ -296,9 +296,7 @@
             name = record['name']

         url = self.API_TEMPLATE + self.RECORDS_TYPE_NAME.format(domain=domain, type=record_type, name=name)
-        records = []
-        records.append(record)
-        self._put(url, json=records)
+        self._put(url, json=record)
         self.logger.info(
             'Updated record. Domain {} name {} type {}'.format(domain, str(record['name']), str(record['type'])))

**Comment by qigtang** [comment]
```diff
--- /usr/local/lib/python2.7/dist-packages/godaddypy/client.py  2018-04-20 22:03:13.466927835 -0700
+++ /usr/local/lib/python2.7/dist-packages/godaddypy/client.py.bak      2018-04-20 21:58:39.783297049 -0700
@@ -296,9 +296,7 @@
             name = record['name']

         url = self.API_TEMPLATE + self.RECORDS_TYPE_NAME.format(domain=domain, type=record_type, name=name)
-        records = []
-        records.append(record)
-        self._put(url, json=records)
+        self._put(url, json=record)
         self.logger.info(
             'Updated record. Domain {} name {} type {}'.format(domain, str(record['name']), str(record['type'])))

```

**Comment by qigtang** [comment]
```diff
--- /usr/local/lib/python2.7/dist-packages/godaddypy/client.py.bak      2018-04-20 21:58:39.783297049 -0700
+++ /usr/local/lib/python2.7/dist-packages/godaddypy/client.py  2018-04-20 22:03:13.466927835 -0700
@@ -296,7 +296,9 @@
             name = record['name']

         url = self.API_TEMPLATE + self.RECORDS_TYPE_NAME.format(domain=domain, type=record_type, name=name)
-        self._put(url, json=record)
+        records = []
+        records.append(record)
+        self._put(url, json=records)
         self.logger.info(
             'Updated record. Domain {} name {} type {}'.format(domain, str(record['name']), str(record['type'])))

```

**Comment by josteink** [comment]
@qigtang you should submit that as a PR. Much easier for @eXamadeus to merge :)

**Comment by siberian1967** [comment]
@qigtang Tested here with le-godaddy-dns and the fix works great. Thank you thank you thank you!

**Comment by pickeld** [comment]
same problem here. please, `pip install --upgrade godaddypy` wont update the lib
what should  i do

**Comment by siberian1967** [comment]
Its probably not PIP available yet. Best bet is to check out the repo and replace your local file with the new version.

**Comment by pickeld** [comment]
>   File "C:\Python27\lib\site-packages\godaddypy\client.py", line 300
>     records = []
>     ^
> IndentationError: unexpected indent
my indentation is correct 
![image](https://user-images.githubusercontent.com/13360982/39202855-db272a46-47fb-11e8-8008-39ee5c3fa3fb.png)

**Comment by siberian1967** [comment]
Python is space sensitive, just put spaces before records = [] until it lines up with the rest of the text, don't tab.

**Comment by pickeld** [comment]
@siberian1967 , thank you. that did the trick. 
will the fix ever be on pip? im using in inside a docker and its a problem for me to change it manually.

**Comment by siberian1967** [comment]
Eventually when there is a formal release.

**Comment by pickeld** [comment]
thank you!
have a great day

**Comment by josteink** [comment]
The fix surely won’t be on pip before @eXamadeus accepts the proposed patch and merges it. 

Right now the latest version *is* based on the latest source, and the source is buggy :)

**Comment by eXamadeus** [comment]
Hey all, sorry life and work has been crazy.  This is top of my priority list.  Expect a proper release by the weekend, if not sooner.  Thanks for the patience!

**Comment by eXamadeus** [comment]
Fixed in #4 and released in 2.2.6

**Comment by josteink** [comment]
Awesome. Thanks!
